tests/test_backup.py

```diff
@@ -743,6 +743,80 @@ def test_cron_retention_only_deletes_OBSOLETE_backups(
+    def test_cron_retention_only_deletes_OBSOLETE_backups_with_incrementals(
+        self, get_available_backups, delete_backup, tmpdir
+    ):
+        """
+        Verify only backups with retention status OBSOLETE are deleted by
+        retention policy.
+        """
+        backup_manager = build_backup_manager()
+        backup_manager.server.config.name = "TestServer"
+        backup_manager.server.config.barman_lock_directory = tmpdir.strpath
+        backup_manager.server.config.backup_options = []
+        backup_manager.server.config.retention_policy = Mock()
+        retention_policies = {
+            "keep_full_backup": BackupInfo.KEEP_FULL,
+            "keep_standalone_backup": BackupInfo.KEEP_STANDALONE,
+            "valid_backup": BackupInfo.VALID,
+            "none_backup": BackupInfo.NONE,
+            "obsolete_backup": BackupInfo.OBSOLETE,
+            "potentially_obsolete_backup": BackupInfo.POTENTIALLY_OBSOLETE,
+        }
+        # Create all available backup and add one child to each
+        available_backups = dict(
+            (
+                key,
+                build_test_backup_info(
+                    server=backup_manager.server,
+                    backup_id=key,
+                    children_backup_ids=[f"child_{index}"],
+                ),
+            )
+            for index, key in enumerate(retention_policies)
+        )
+        # Create one child per available backup
+        available_children = dict(
+            (
+                f"child_{index}",
+                build_test_backup_info(
+                    server=backup_manager.server,
+                    backup_id=f"child_{index}",
+                    parent_backup_id=key,
+                ),
+            )
+            for index, key in enumerate(retention_policies)
+        )
+        # Update available backups
+        available_backups.update(available_children)
+        get_available_backups.return_value = available_backups
+
+        children_retention_policies = {
+            "child_0": BackupInfo.VALID,
+            "child_1": BackupInfo.VALID,
+            "child_2": BackupInfo.VALID,
+            "child_3": BackupInfo.NONE,
+            "child_4": BackupInfo.OBSOLETE,
+            "child_5": BackupInfo.POTENTIALLY_OBSOLETE,
+        }
+        # Update retention_policies with children status
+        retention_policies.update(children_retention_policies)
+        backup_manager.config.retention_policy.report.return_value = retention_policies
+
+        backup_manager.cron_retention_policy()
+
+        delete_backup.call_count == 2
+        expected_calls = [
+            call(available_backups["child_4"], skip_wal_cleanup_if_standalone=False),
+            call(
+                available_backups["obsolete_backup"],
+                skip_wal_cleanup_if_standalone=False,
+            ),
+        ]
+        delete_backup.assert_has_calls(expected_calls, any_order=False)
+
```

Comment:
I added this because cron_retetion_policy() calls the report() method. So i wanted to be sure that the child was deleted with its parent calling the delete twice. But now RE-reading the name of the test, i am thinking over what you pointed out. Maybe we could add the children parts to the existing test?

Reply:
What I mean is: `cron_retention_policy` doesn't know which type of backup it is dealing with:

* Is it rsync?
* Is it Postgres 16 or older?
* Is it Postgres 17 full?
* Is it Postgres 17 incremental?

That information doesn't matter for that method. The only thing that matters is:

* Is the backup `OBSOLETE`?

The logic which you mentioned belongs to `_backup_report`. That's the method we should check if it is performing as expected.

I mean, make sure `_backup_report` is marking the children nodes in the tree with the expected status.

In this case, the test on `cron_retention_policy` seems redundante because we are artificially mocking everything, including the return of `report` method.

The idea is that we get to something like this:

* When testing `_backup_report`: assuming I have this set of full and incremental backups, I would expect backups X, Y and Z to be marked as status "SOME STATUS";
* When testing `cron_retention_policy`: assuming `report` gives me this set of backups with their status, I remove backups X, Y and Z.

Does that make it clearer what I have in mind?

Reply:
> When testing cron_retention_policy: assuming report gives me this set of backups with their status, I remove backups X, Y and Z.

FWIW this is already covered by tests like `test_cron_retention_only_deletes_OBSOLETE_backups`.

Reply:
Yeah seems not necessary to me